pkg/controller/direct/monitoring/dashboard_mappings.go

```diff
@@ -40,6 +40,27 @@ func Empty_ToProto(mapCtx *MapContext, in *krm.Empty) *emptypb.Empty {
 	return out
 }
 
+func AlertChart_FromProto(mapCtx *MapContext, in *pb.AlertChart) *krm.AlertChart {
+	if in == nil {
+		return nil
+	}
+	out := &krm.AlertChart{}
+	if in.Name != "" {
+		out.AlertPolicyRef = &krm.MonitoringAlertPolicyRef{
+			External: in.Name,
+		}
+	}
+	return out
+}
+func AlertChart_ToProto(mapCtx *MapContext, in *krm.AlertChart) *pb.AlertChart {
+	if in == nil {
+		return nil
+	}
+	out := &pb.AlertChart{}
+	out.Name = in.AlertPolicyRef.External
```

Comment:
Do we need to check the null pointer `in.AlertPolicyRef` here? (I hit a similar issue but use a different auto proto method)

Reply:
Technically no, because the field is required.  But I agree with you, we should be checking it from a defensive point of view.  Fixed!